### ISSUE TITLE
Bug: Allow to access plugin meta data for all loaded plugins again

### ISSUE THREAD

**Issue by swernerx** [eslint/eslint#15546]
### Environment

Node version: 16.13.1
npm version: 8.1.2
Local ESLint version: 8.7.0
Global ESLint version: -
Operating System: MacOS

### What parser are you using?

@typescript-eslint/parser

### What did you do?

Previously, with ESLint v7 and CLIEngine it was possible to load the rules by `getRules()`. As far as I understood this API is now gone because it had some unwanted side-effects. So far so good.

I'd like to point out that there were a few useful uses of this API that are unfortunately no longer supported currently. We have written an OSS tool that runs Prettier and ESLint in combination. The idea is that the tool only executes rules that are "fixable" - so it filters the user's ESLint rules, so to speak. This way you can realize a kind of post-processing of the ruleset of Prettier without too many performance losses. This worked really well with v7. 

Despite some research, I don't see any possibility of a similar API. I'm not only concerned about internal rules but also those loaded by plugins (React, JSX, ...).

My current approach was:

```ts
async function getRulesHelper() {
  const root = (await pkgDir()) as string

  const instance = new ESLint({
    cwd: root,
    useEslintrc: false,
    overrideConfig: {
      plugins: ["react"],
      rules: {
        "react/destructuring-assignment": ["error", "always"]
      }
    }
  })
  const config = await instance.calculateConfigForFile(
    path.resolve(root, "index.tsx")
  )

  const linter = new Linter({ cwd: root })
  const result = await linter.verify("function hello() { return 42 }", config, {
    filename: "index.tsx"
  })
  console.log("Result:", result)

  const rules = linter.getRules()
  console.log("ALL:", rules.size)
}
```

Do you have any hints for solving this requirement with v8?

### What did you expect to happen?

That's the desired output:

```
Result: []
ALL: 382 (286 eslint + 96 react plugin)
```

### What actually happened?

That's the actual output:

```
Result: [
  {
    ruleId: 'react/destructuring-assignment',
    message: "Definition for rule 'react/destructuring-assignment' was not found.",
    line: 1,
    column: 1,
    endLine: 1,
    endColumn: 2,
    severity: 2,
    nodeType: null
  }
]
ALL: 286
```

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by swernerx** [comment]
I made some further progress and would suggest this class method to be added to the new `ESLint` class:

```js
    /**
     * Returns a clone of relevant meta data for all loaded rules.
     * @returns {Object} A mapping of ruleIds to rule meta objects.
     */
    getLoadedRulesMeta() {
        const { cliEngine } = privateMembersMap.get(this);
        const rules = cliEngine.getRules()
        const meta = new Map();
        for (const [ruleId, rule] of rules) {
            meta.set(ruleId, {
                ...rule.meta.docs,
                fixable: rule.meta.fixable || false,
                hasSuggestions: rule.meta.hasSuggestions || false
            })
        }
        return meta
    }
```

If you realize that this is a good adaptation - which I hope it is - then feel free to let me know and I'll deliver the PR afterwards.

**Comment by swernerx** [comment]
It should be safe as it does offer all data on a deep data clone and omits all references to internal data structures.

**Comment by swernerx** [comment]
Anyway... here is the pull request: https://github.com/eslint/eslint/pull/15547

**Comment by nzakas** [comment]
Yes, we intentionally removed this capability because the method returned different data depending on when you called it. We aren’t going to add back any method that would behave in a similar way.

As mentioned in the [migration guide](https://eslint.org/docs/user-guide/migrating-to-8.0.0#the-cliengine-class-has-been-removed), you can use the builtinRules map from `eslint/use-at-your-own-risk` to get internal rule meta data and you can just directly load plugins using `require()‘ to get that meta data.
